### ISSUE TITLE
Cannot apply local values.yaml to deployed Helm chart

### ISSUE THREAD

**Issue by tolko** [argoproj/argo-cd#1413]
**Describe the bug**
getting the following error:
rpc error: code = Unknown desc = Error: open path/to/values.yaml: no such file or directory
while using 
argocd app set helm-app --values values.yaml

**To Reproduce**
1. Add repository to argocd
2. add application
3. try to set values yaml to newly created application with:
argocd app set helm-app --values path/to/values.yaml
4. See error in UI for this specific application

**Expected behavior**
is to add entire values yaml to this application instead of using workaround like setting parameters one by one

**Comment by alexec** [comment]
Also discussed in the Slack channel.  AFAIK, this needs to be the path to the values file within the Git repo.

**Comment by tolko** [comment]
@alexec maybe documentation change is required...
https://argoproj.github.io/argo-cd/user-guide/application_sources/#values-files

Can you recall any suitable way to pass locally stored values.yaml to the application?

**Comment by alexec** [comment]
I think you're probably right about the docs, but this is work in progress. Have you considered using parameters?

https://argoproj.github.io/argo-cd/user-guide/application_sources/#helm-parameters

**Comment by tolko** [comment]
yes, currently this is my workaround but it will be a lot straight forward if I can use locally stored values yaml. 
or if this yaml is stored on the same cluster or something
I am open to ideas :)

**Comment by sahil87** [comment]
Using parameters in certain cases isn't very straight forward and extremely lengthy. Eg:
* Ingress annotation need to be escaped properly.
   ```yaml
    parameters:
      - name: ingress.annotations.ingress\.kubernetes\.io/protocol
        value: "http"
   ```
   This isn't obvious and is tough to figure out.
* My values.yaml for traefik/mysql/mattermost etc (and many other charts) are 2/3 pages long. Moving them within `helm: parameters: []` makes the application.yaml super long and difficult to manage.
   
   Not using helm and switching to Kustomize also isn't desirable as then one also need to worry about ensuring whether Statefulsets/PVCs etc were properly setup. Using helm (locked on to a targetRevision) allows one the peace of mind of the community having tested out the charts.

Keeping values.yaml in the same repo as the chart isn't always desirable. Keeping values.yaml in a separate repo than the charts will definitely make my life a lot easier. argo-cd is an awesome tool, and having this feature will make integration of argo-cd into my CI flow a lot simpler than it is now.

**Comment by dimitarKiryakov** [comment]
+1 for this issue.
It is very complicated to parse the values yaml's. Passing parameters directly is not okay for a workaround, as it is a big overhead that could cause a lot of issues when working with big yaml files. Please consider a way to specify a values yaml file at least from a different repo.

**Comment by evrardjp** [comment]
I am in the same case. I would like to even propose a "curated list of apps" where I propose my own overrides of helm charts, all of that into the Application. A user can then fork my repository, adapt to his needs, but would still have the "easy" way to override my "curated list of apps". I would like something which does some kind of "pass through" of files into the application.

This brings me a few questions:
* Why is the formatting of "parameters" written this way?
* Why do all the parameters need to be strings? (Check also this: https://gist.github.com/evrardjp/886e3143526a639feeae0687dd4bda03) -- This might help understand why not passing things.
* Were there any alternatives to "parameters" proposed in the past?
* If we can't pass files directly next to the "Application" file, did we think of an upload of the files to a configmap, which would then be read inside the Application definition?

**Comment by alexec** [comment]
I believe that a recent PR allows you to sync and app using a local directory, but we don't support local values files (this would not work for Kustomize for example). Open for contribuiton.

**Comment by tolko** [comment]
Hi,
maybe this issue can be part of 1.3 Helm Milestone?
and also can be enhanced by allowing configuration (additional values files) to reside in a private git repository without the need to cloning the chart in that repository.

**Comment by crenshaw-dev** [comment]
I think this is a duplicate of https://github.com/argoproj/argo-cd/issues/2789 (this one predates the other, but the other has become the primary tracker for this problem). Will close as duplicate, and someone can ping me if they disagree.  :-)
